tests/core/test_evaluation_parameters.py

```diff
@@ -118,6 +118,51 @@ def test_parse_evaluation_parameter():
     )
 
 
+def test_query_store_results_in_evaluation_parameters(data_context_with_query_store):
+    TITANIC_ROW_COUNT = 1313  # taken from the titanic db conftest
+    DISTINCT_TITANIC_ROW_COUNT = 4
+
+    test_cases = [
+        {
+            "context": "parse_evaluation_parameters correctly resolves a stores URN",
+            "parameter_expression": "urn:great_expectations:stores:my_query_store:col_count",
+            "res_func": lambda x: x == TITANIC_ROW_COUNT,
+        },
+        {
+            "context": "and can handle an operator",
+            "parameter_expression": "urn:great_expectations:stores:my_query_store:col_count * 2",
+            "res_func": lambda x: x == TITANIC_ROW_COUNT * 2,
+        },
+        {
+            "context": "can even handle multiple operators",
+            "parameter_expression": "urn:great_expectations:stores:my_query_store:col_count * 0 + 100",
+            "res_func": lambda x: x == 100,
+        },
+        {
+            "context": "allows stores URNs with functions",
+            "parameter_expression": "cos(urn:great_expectations:stores:my_query_store:col_count)",
+            "res_func": lambda x: math.isclose(math.cos(TITANIC_ROW_COUNT), x),
+        },
+        {
+            "context": "multiple stores URNs can be used",
+            "parameter_expression": "urn:great_expectations:stores:my_query_store:col_count - urn:great_expectations:stores:my_query_store:dist_col_count",
+            "res_func": lambda x: x == TITANIC_ROW_COUNT - DISTINCT_TITANIC_ROW_COUNT,
+        },
+        {
+            "context": "complex expressions can combine operators, urns, and functions",
+            "parameter_expression": "abs(-urn:great_expectations:stores:my_query_store:col_count - urn:great_expectations:stores:my_query_store:dist_col_count)",
+            "res_func": lambda x: x == TITANIC_ROW_COUNT + DISTINCT_TITANIC_ROW_COUNT,
+        },
+    ]
+    for case in test_cases:
```

Comment:
I love the idea of the parameterized test cases, but I'm finding them somewhat difficult to read and parse. Part of this is that I'm not clear on what `parse_evaluation_parameter` is returning for each of these. Could you either break them out to more clearly indicate how the assertions are working, or add some more documentation as to how this test is operating?

Reply:
yeah, i debated between doing this explicitly or in a loop, and eventually decided on the loop. if you feel strongly about the style i'm willing to change it.

`parse_evaluation_parameter` is returning a scalar value, and the lambda functions allow us to describe in code what we expect that scalar to be. We could also just write `assert <func> == 100`, but since we're evaluating complex statements, I like that that the expected result mirrors the shape of the Evaluation Parameter.

Reply:
refactored to address